runtime/src/server/middleware/get_page_handler.ts

```diff
@@ -346,7 +346,7 @@ export function get_page_handler(
 					const legacy_main = `${req.baseUrl}/client/legacy/${build_info.legacy_assets.main}`;
 					script += `(function(){try{eval("async function x(){}");var main="${main}"}catch(e){main="${legacy_main}"};var s=document.createElement("script");try{new Function("if(0)import('')")();s.src=main;s.type="module";s.crossOrigin="use-credentials";}catch(e){s.src="${req.baseUrl}/client/shimport@${build_info.shimport}.js";s.setAttribute("data-main",main);}document.head.appendChild(s);}());`;
 				} else {
-					script += `var s=document.createElement("script");try{new Function("if(0)import('')")();s.src="${main}";s.type="module";s.crossOrigin="use-credentials";}catch(e){s.src="${req.baseUrl}/client/shimport@${build_info.shimport}.js";s.setAttribute("data-main","${main}")}document.head.appendChild(s)`;
```

Comment:
what was this `try` / `catch` trying to detect? I feel like it was to see what browser you were using and can't simply be removed

Reply:
Looks like it was trying to detect "import" support to add the module script, otherwise it added "shimport".

Reply:
If there's a way to detect ESM/`import`/`import()` support without a try-catch, then I'd be all for switching to that. Otherwise, as I indicated below, I don't think we can just do this without it being a breaking change. (And it'd be a breaking change that I don't think could be easily worked around for people that need to support these browsers. They'd be stuck on the old version.)

Reply:
@Conduitry Agreed. I'll have a look 👍

Reply:
Actually, can't we use `nomodule` for backwards compat?